third_party/nvidia/backend/compiler.py

```diff
@@ -204,6 +204,7 @@ def make_ttgir(mod, metadata, opt, capability):
         # TTIR -> TTGIR
         pm = ir.pass_manager(mod.context)
         pm.enable_debug()
+        passes.ttir.add_loop_unroll(pm)
```

Comment:
sounds like it should belong to make_ttir? Seems odd to make it part of ttgir but run before conversion to ttgir

Reply:
Yeah I was placing it there but didn't figure it out how to make it work with kernel override. The current placement allows to annotate ttir with the unroll factor and continue the compilation for ttgir, with kernel override. Perhaps we do not want the override with https://github.com/triton-lang/triton/pull/4662 . WDYT?

Reply:
ah I see, well you shouldn't need to rely on override long term. It only makes sense to have this pass if we expose it at the front end level

Reply:
Sounds good. Moved it back.